File: pset6/caesar.py
import cs50
import sys

def main():

    # Check that user passed one command-line argument
    if len(sys.argv) != 2:
        print("Please pass the key as unique command-line argument")
        exit()
    else:
        # Get the key and convert it to an integer
        k = int(sys.argv[1])

        # Prompt for plain text to the user
        print("plaintext:", end=" ")
        p = cs50.get_string()
        # cipher text
        ciphertext = []

        # Cipher plaintext character by key
        for c in p:
            if c.isalpha():
                ciphertext.append(caesar(c, k))
            else:
                ciphertext.append(c)

        # print ciphertext
        print("ciphertext: " + "".join(ciphertext))

# caesar cipher preserving key function
def caesar (c, k):
    if c.isupper():
        return chr(((ord(c) - 65 + k) % 26) + 65)
    else:
        return chr(((ord(c) - 97 + k) % 26) + 97)

if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

// Get the key from command-line argument
int main(int argc, string argv[])
{
    // Check that user passed one command-line argument
    if (argc != 2)
    {
        printf("Please pass one, and only one, command-line argument that should be a non-negative integer.\n");
        return 1;
    }

     // Get the key and convert it to an integer
    int k = atoi(argv[1]);

    // Prompt for plain text to the user
    printf ("plaintext: ");
    string p = get_string();
    string c = p;

    // for each character in the plaintext string
    for (int i = 0, n = strlen(p); i < n; i++)
    {
        // if alphabetic
        if (isalpha(p[i]))
        {
            // Preserve case
            // Cipher plaintext character by key
            if (isupper(p[i]))
            {
                c[i] = ((p[i] - 65 + k) % 26) + 65;
            }

            if (islower(p[i]))
            {
                c[i] = ((p[i] - 97 + k) % 26) + 97;
            }
        }
    }
    // Print ciphertext
    printf("ciphertext: %s\n", c);
}